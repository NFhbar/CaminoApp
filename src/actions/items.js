import uuid from 'uuid';

// ADD_ITEM
export const addItem = (
  {
    description = '',
    note = '',
    amount = 0,
    createdAt = 0
  } = {}
) => ({
  type: 'ADD_ITEM',
  item: {
    id: uuid(),
    description,
    note,
    amount,
    createdAt
  }
});

// REMOVE_ITEM
export const removeItem = ({ id } = {}) => ({
  type: 'REMOVE_ITEM',
  id
});

// EDIT_ITEM
export const editItem = (id, updates) => ({
  type: 'EDIT_ITEM',
  id,
  updates
});
