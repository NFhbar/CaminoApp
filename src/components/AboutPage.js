import React from 'react';
import { NavLink } from 'react-router-dom';
import { row, column } from 'reactstrap';

const AboutPage = () => (
  <div className="row">
    <div className="col-sm-12">
      <p>From About</p>
    </div>
  </div>
);

export default AboutPage;
