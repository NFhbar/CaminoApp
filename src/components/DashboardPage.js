import React from 'react';
import { row, column } from 'reactstrap';
import ItemList from './ItemList';
import ItemListFilters from './ItemListFilters';

const DashboardPage = () => (
  <div className="row">
    <div className="col-sm-12">
      <ItemListFilters />
      <ItemList />
    </div>
  </div>
);

export default DashboardPage;
