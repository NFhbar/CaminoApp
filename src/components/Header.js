import React from 'react';
import { NavLink } from 'react-router-dom';
import { row, column } from 'reactstrap';

const Header = () => (
  <div className="row">
    <div className="col-sm-12">
      <h1>CaminoApp</h1>
      <NavLink to="/" activeClassName="is-active" exact={true}>Dashboard</NavLink>
      <NavLink to="/create" activeClassName="is-active">Create Item</NavLink>
      <NavLink to="/about" activeClassName="is-active">About</NavLink>
    </div>
  </div>
);

export default Header;
