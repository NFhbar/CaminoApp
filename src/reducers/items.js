// Items Reducer

const itemsReducerDefaultState = [];

export default (state = itemsReducerDefaultState, action) => {
  switch (action.type) {
    case 'ADD_ITEM':
      return [
        ...state,
        action.item
      ];
    case 'REMOVE_ITEM':
      return state.filter(({ id }) => id !== action.id);
    case 'EDIT_ITEM':
      return state.map((item) => {
        if(item.id === action.id){
          return {
            ...item,
            ...action.updates
          };
        } else {
          return item;
        };
      });
    default:
      return state;
  }
};
