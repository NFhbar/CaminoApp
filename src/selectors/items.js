import moment from 'moment';

// Get visible items

export default (items, { text, sortBy, startDate, endDate}) => {
  return items.filter((item) => {
    const createdAtMoment = moment(item.createdAt);
    const startDateMatch = startDate ? startDate.isSameOrBefore(createdAtMoment,'day'): true ;
    const endDateMatch = endDate ? endDate.isSameOrAfter(createdAtMoment, 'day'): true;
    const textMatch = item.description.toLowerCase().includes(text.toLowerCase());

    return startDateMatch && endDateMatch && textMatch;
  }).sort((a, b) => {
    if(sortBy === 'date') {
      return a.createdAt < b.createdAt ? 1 : -1;
    } else if (sortBy === 'amount') {
      return a.amount < b.amount ? 1: -1;
    }
  });
};
