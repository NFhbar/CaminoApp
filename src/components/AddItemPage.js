import React from 'react';
import { connect } from 'react-redux';
import { row, column } from 'reactstrap';
import ItemForm from './ItemForm'
import { addItem } from '../actions/items';

export class AddItemPage extends React.Component {
  onSubmit = (item) => {
    this.props.addItem(item);
    this.props.history.push('/');
  };
  render() {
    return (
      <div className="row">
        <div className="col-sm-12">
          <h1>Add Item</h1>
          <ItemForm
            onSubmit={this.onSubmit}
          />
        </div>
      </div>
    );
  }
}

const mapDispatchToProps = (dispatch) => ({
  addItem: (item) => dispatch(addItem(item))
});

export default connect(undefined, mapDispatchToProps)(AddItemPage);
