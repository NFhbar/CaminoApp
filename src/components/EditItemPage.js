import React from 'react';
import { connect } from 'react-redux';
import { row, column } from 'reactstrap';
import ItemForm from './ItemForm';
import { editItem, startRemoveItem } from '../actions/items';

export class EditItemPage extends React.Component {
  onSubmit = (item) => {
    this.props.editItem(this.props.item.id, item);
    this.props.history.push('/');
  };
  onRemove = () => {
    this.props.startRemoveItem({ id: this.props.item.id });
    this.props.history.push('/');
  };
  render() {
    return (
      <div className="row" id="edit-item-page">
        <div className="col-sm-12">
          <ItemForm
            item={this.props.item}
            onSubmit={this.onSubmit}
          />
          <button type="submit" id="btn-remove" className="btn btn-md btn-danger" onClick={this.onRemove}>Remove Item</button>
        </div>
      </div>
    );
  }
};

const mapStateToProps = (state, props) => ({
  item: state.items.find((item) => item.id === props.match.params.id)
});

const mapDispatchToProps = (dispatch, props) => ({
  editItem: (id, item) => dispatch(editItem(id, item)),
  startRemoveItem: (data) => dispatch(startRemoveItem(data))
});

export default connect(mapStateToProps, mapDispatchToProps)(EditItemPage);
