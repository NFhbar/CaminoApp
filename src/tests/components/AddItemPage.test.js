import React from 'react';
import { shallow } from 'enzyme';
import { AddItemPage } from '../../components/AddItemPage';
import items from '../fixtures/items';

let addItem, history, wrapper;

beforeEach(() => {
  addItem = jest.fn();
  history = { push: jest.fn() };
  wrapper = shallow(<AddItemPage addItem={addItem} history={history} />);
});

test('should render AddItemPage correctly', () => {
  expect(wrapper).toMatchSnapshot();
});

test('should handle onSubmit', () => {
  wrapper.find('ItemForm').prop('onSubmit')(items[1]);
  expect(history.push).toHaveBeenLastCalledWith('/');
  expect(addItem).toHaveBeenLastCalledWith(items[1]);
});
