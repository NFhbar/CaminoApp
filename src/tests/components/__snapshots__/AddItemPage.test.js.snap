// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render AddItemPage correctly 1`] = `
<div
  className="row"
>
  <div
    className="col-sm-12"
  >
    <h1>
      Add Item
    </h1>
    <ItemForm
      onSubmit={[Function]}
    />
  </div>
</div>
`;
